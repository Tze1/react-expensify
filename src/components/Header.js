import React from 'react';
import { connect } from 'react-redux';
import { startLogout } from '../actions/authActions';
import { NavLink } from 'react-router-dom';

export const Header = (props) => (
  <header className="header component">
    <h1 className="header-title">React-Expensify</h1>
    <div className="header-nav">
      <NavLink to="/dashboard" activeClassName="is-active">Dashboard</NavLink> |&nbsp;
      <NavLink to="/add" activeClassName="is-active">Add Expense</NavLink>
    </div>
    <div className="header-auth">
      <button className="header-auth-btn btn exit" onClick={props.startLogout}>Log out</button>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => { dispatch(startLogout()); }
});

export default connect(undefined, mapDispatchToProps)(Header);
