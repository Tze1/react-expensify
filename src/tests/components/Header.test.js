import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header.js';

test('Should render header correctly', () => {
  const wrapper = shallow(<Header startLogout={() => {}} />);

  expect(wrapper).toMatchSnapshot();
});

test('Should fire startLogout properly', () => {
  const startLogout = jest.fn();
  const wrapper = shallow(<Header startLogout={startLogout} />);

  wrapper.find('.header-auth-btn.exit').simulate('click');
  expect(startLogout).toHaveBeenCalled();
});
