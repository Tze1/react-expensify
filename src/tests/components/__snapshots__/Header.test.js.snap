// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render header correctly 1`] = `
<header
  className="header component"
>
  <h1
    className="header-title"
  >
    React-Expensify
  </h1>
  <div
    className="header-nav"
  >
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      to="/dashboard"
    >
      Dashboard
    </NavLink>
     |Â 
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      to="/add"
    >
      Add Expense
    </NavLink>
  </div>
  <div
    className="header-auth"
  >
    <span
      className="header-auth-greeting"
    >
      Joe Blow
    </span>
    <button
      className="header-auth-btn button exit"
      onClick={[Function]}
    >
      Log out
    </button>
  </div>
</header>
`;
